handler/src/main/java/io/netty/handler/ssl/Conscrypt.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.util.internal.PlatformDependent;
+
+import javax.net.ssl.SSLEngine;
+import java.lang.reflect.Method;
+
+/**
+ * Contains methods that can be used to detect if conscrypt is usable.
+ */
+final class Conscrypt {
+    // This class must ensure it not try to load any conscrypt classes without using reflection to also allow
```

Comment:
Consider rewording and adding to class level javadocs:

This class exists to avoid loading other conscrypt related classes using features only available in JDK8+, because we need to maintain JDK6+ runtime compatibility and JDK7+ compile compatibility.